### ISSUE TITLE
Install & Update fail at "clone github to /etc/.pihole" step

### ISSUE THREAD

**Issue by mholmes10** [pi-hole/pi-hole#2357]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**
2

---
**Expected behaviour:**
Updating pihole to the latest version or clean install
This step should clone the github but it errors out

**Actual behaviour:**
Error: Could not update local repository. contact support

**Steps to reproduce:**

`{Detailed steps of how we can reproduce this}`
I initially used pihole -up  (but it errored out and left me with no working dns)
then I used pihole uninstall, reboot and then curl -sSL https://install.pi-hole.net |bash

It errors out after the steps to set up setting (ip, forward dns, logs, etc) and moves to the step
Check for existing repository in /etc/.pihole
Clone https://github.com/pi-hole/pi-hole.git into /etc/.pihole
ERROR: Could not update local repository. Contact Support

**Troubleshooting undertaken, and/or other relevant information:**

I tired to clone the repository manually with 
"git clone https://github.com/pi-hole/pi-hole.git"

I get a few status messages (cloning, remote, receiving objs) 
then after getting "Resolving deltas:100% (8147/8147), done"
I get a Segmentation fault
and an error message "git: 'rev-liSt' is not a git command.  See 'git --help'.
Did you mean this?
    rev-list

It looks like the error may be in the github clone commands.

thanks

**Comment by dschaper** [comment]
I've looked through ht code base and I haven't been able to find any calls to `git rev-list` or `git rev-liSt`. Does this problem still occur when you remove the `/etc/.pihole` directory and `git clone https://github.com/pi-hole/pi-hole.get /etc/.pihole`?

**Comment by cholcombe973** [comment]
Looks to be a permissions issue on my pi.  If you try to sudo clone the repo it gives you an error message about root git being symlinked and tells you to just sudo rm the link and you're good to go.

**Comment by dschaper** [comment]
Is this OctoPi?

**Comment by cholcombe973** [comment]
It was until i deleted it haha.

**Comment by dschaper** [comment]
Yes, there's a bit of a difference in opinions on the use of `git` and `sudo`. But we hope to have something soon that eliminates the issue completely. Let us know if there are any other issues.

**Comment by mholmes10** [comment]
yes I get the same error when I run 
git clone https://github.com/pi-hole/pi-hole.git /etc/.pihole

Cloning into '/etc/.pihole'...
remote: Counting objects: 13203, done.
remote: Compressing objects: 100% (66/66), done.
remote: Total 13203 (delta 45), reused 26 (delta 12), pack-reused 13125
Receiving objects: 100% (13203/13203), 3.95 MiB | 3.52 MiB/s, done.
Resolving deltas: 100% (8154/8154), done.
git: 'rev-liSt' is not a git command. See 'git --help'.

Did you mean this?
	rev-list
Segmentation fault

I tried running it to another directory and got the following error

$ git clone https://github.com/pi-hole/pi-hole.git ~/.pihole
Cloning into '/home/pi/.pihole'...
remote: Counting objects: 13203, done.
remote: Compressing objects: 100% (66/66), done.
remote: Total 13203 (delta 45), reused 26 (delta 12), pack-reused 13125
Receiving objects: 100% (13203/13203), 3.95 MiB | 3.77 MiB/s, done.
Resolving deltas: 100% (8154/8154), done.
Segmentation fault
git: 'rev-liSt' is not a git command. See 'git --help'.

Did you mean this?
	rev-list

**Comment by dschaper** [comment]
What does `git clone https://github.com/pi-hole/AdminLTE.git /tmp/adminlte` do. I think there is something amiss with your OS install. Those are bare `git` calls and do not execute any scripts or code from the cloned repositories. You can try any `git` repository other than a Pi-hole one and see if the behavior is the same.

**Comment by GuillaumeRossolini** [comment]
For anyone else still finding their way here as I did, it may be because you aborted or restarted the install.

If that happens, DNS name resolution may be broken, and the first step in the install process where this comes up is at the git cloning step, which gives the same error message:
`Error: Could not update local repository. contact support`

You can try the following command:
`ping google.com`

If this command fails, then another file to reset is `/etc/resolv.conf` which points to 127.0.0.1 as a name server, but since the install hasn't finished, this line won't work. You can temporarily change this IP to any DNS server of your choice, for example 1.1.1.1, then install/update pi-hole again, and at the end make sure this file points to 127.0.0.1 again.

**Comment by bit4us** [comment]
Hi guys, 
I have encountered the same issue on a OctoPi installed Pi. As a first attempt the installer got stuck and I had to unplug the power cable to make my PI work again. At the second attempt I got the same error like all of you. In order to fix it I had to allow git to run as sudo / root (I know it's a big no-no, but it was the only thing that made it work) because /etc/.pihole was a root directory. So I did: 
`sudo rm /root/bin/git `
in order to allow the installer to make the git clone. After that, the installer worked like a charm, and after a system reboot I was able to use the software.

I hope it will resolve the issues for many people this solution.

Best regards.
